executor/load_data.go

```diff
@@ -129,6 +129,36 @@ type FieldMapping struct {
 	UserVar *ast.VariableExpr
 }
 
+// reorderColumns reorder the e.insertColumns according to the order of columnNames
+// Note: We must ensure there must be one-to-one mapping between e.insertColumns and columnNames in terms of column name.
+func (e *LoadDataInfo) reorderColumns(columnNames []string) error {
+	cols := e.insertColumns
+
+	if len(cols) != len(columnNames) {
+		return ErrColumnsNotMatched
+	}
+
+	reorderedColumns := make([]*table.Column, len(cols))
+
+	if columnNames == nil {
+		return nil
+	}
+
+	mapping := make(map[string]int)
+	for idx, colName := range columnNames {
+		mapping[colName] = idx
+	}
+
+	for _, col := range cols {
+		idx := mapping[col.Name.O]
```

Comment:
Should use `col.Name.L`? Also in line#149 should use `strings.ToLower(colName)` as a key?

Please add a test for this case.

Reply:
Good catch, we should convert them to lower case. I will modify it immediately.